src/controllers/router-store/index.tsx

```diff
@@ -366,6 +345,33 @@ const actions: AllRouterActions = {
         history[updateType](updatedRelativePath);
       }
     },
+  loadRoute:
+    () =>
+    ({ getState }) => {
+      const { loader, match, query, route } = getState();
+
+      return loader.load({ match, query, route });
+    },
+  prefetchRoute:
+    (path, nextContext) =>
+    ({ getState }) => {
+      const { loader, routes, basePath, onPrefetch } = getState();
+
+      if (!nextContext && !isExternalAbsolutePath(path)) {
+        const location = parsePath(getRelativePath(path, basePath) as any);
+        nextContext = findRouterContext(routes, { location, basePath });
+      }
+
+      if (nextContext == null) return;
+      const nextLocationContext = nextContext;
+
+      batch(() => {
+        loader.prefetch(nextLocationContext);
+        // prefetchResources(nextResources, nextLocationContext, {});
+        if (onPrefetch) onPrefetch(nextLocationContext);
```

Comment:
Now I wonder: should this be moved to a loader too? This could very well be a `webpackLoader` and could handle what `onPrefetch` does in Jira without the need of a specific prop 👀

Reply:
Yeah, I think it can too. Jira can easily add a custom loader to maintain parity behaviour. It would be a breaking change though but a minor one at that since there wouldn't be too many consumers.